### PR TITLE
ARROW-3787: [Rust] Implement From<ListArray> for BinaryArray

### PR DESCRIPTION
**Pull request by paddyhoran**

### CODE REVIEW

**Review by sunchao — APPROVED** [review]

File: rust/src/array.rs [hunk 1]
```diff
@@ -585,6 +585,29 @@ impl<'a> From<Vec<&'a str>> for BinaryArray {
     }
 }
 
+/// Creates a `BinaryArray` from `List<u8>` array
+impl From<ListArray> for BinaryArray {
+    fn from(v: ListArray) -> Self {
+        assert_eq!(
+            v.data().child_data()[0].child_data().len(),
+            0,
+            "BinaryArray can only be created from list array of u8 values (i.e. List<PrimitiveArray<u8>>)."
```
Comment (kszucs) [inline rust/src/array.rs]: @paddyhoran Do We have a test for it?

Comment (paddyhoran) [reply rust/src/array.rs]: I didn't want to be overkill on the testing as the logic is relatively simple but I'll add a few more to cover the other constraints, thanks @kszucs

**Comment by paddyhoran** [comment]
@kszucs I think this is ready, just pinging you.

**Review by kszucs — APPROVED** [review]
LGTM

**Comment by kszucs** [comment]
Thanks @paddyhoran @sunchao !

**Pull request closed without merging**